File: ruby/lib/Sikaru/types/trace_import_scope_request.rb
# frozen_string_literal: true

module Sikaru
  module Types
    class TraceImportScopeRequest < Internal::Types::Model
      field :external_ids, -> { Internal::Types::Array[String] }, optional: true, nullable: false, api_name: "externalIds"

      field :since, -> { String }, optional: true, nullable: false

      field :until_, -> { String }, optional: true, nullable: false, api_name: "until"
    end
  end
end

File: python/src/sikaru_api/types/trace_import_scope_request.py

import datetime as dt
import typing

import pydantic
import typing_extensions
from ..core.pydantic_utilities import IS_PYDANTIC_V2, UniversalBaseModel
from ..core.serialization import FieldMetadata


class TraceImportScopeRequest(UniversalBaseModel):
    external_ids: typing_extensions.Annotated[
        typing.Optional[typing.List[str]], FieldMetadata(alias="externalIds"), pydantic.Field(alias="externalIds")
    ] = None
    since: typing.Optional[dt.datetime] = None
    until: typing.Optional[dt.datetime] = None

    if IS_PYDANTIC_V2:
        model_config: typing.ClassVar[pydantic.ConfigDict] = pydantic.ConfigDict(extra="allow", frozen=True)  # type: ignore # Pydantic v2
    else:

        class Config:
            frozen = True
            smart_union = True
            extra = pydantic.Extra.allow
